app/api/files/delete/route.ts

```diff
@@ -0,0 +1,53 @@
+import { NextResponse } from "next/server";
+import supabase from "@/lib/supabase/serverClient";
+import { SUPABASE_STORAGE_BUCKET } from "@/lib/consts";
+
+export async function POST(req: Request) {
+  try {
+    const body = await req.json();
+    const id = String(body.id || "");
+    const storageKey = String(body.storageKey || "");
+    const ownerAccountId = String(body.ownerAccountId || "");
+
+    if (!id || !storageKey || !ownerAccountId) {
+      return NextResponse.json({ error: "Missing id, storageKey or ownerAccountId" }, { status: 400 });
+    }
+
+    // Load file row to check ownership and type
+    const { data: row, error: rowError } = await supabase
+      .from("files")
+      .select("id, owner_account_id, is_directory, storage_key")
+      .eq("id", id)
+      .single();
```

Comment:
Single Responsibility Principle
- actual: supabase libs defined within an API endpoint
- required: supabase libs defined in standalone file within /lib/supabase